doh-server/server.go

```diff
@@ -265,17 +265,25 @@ func (s *Server) findClientIP(r *http.Request) net.IP {
 	if XRealIP != "" {
 		addr := strings.TrimSpace(XRealIP)
 		ip := net.ParseIP(addr)
+		if !s.conf.LocalIPFilter {
+			return ip
+		}
```

Comment:
I think you can rewrite using an "A or B" expression.
```go
if !s.conf.LocalIPFilter || jsondns.IsGlobalIP(ip) {
```

Also it seems that you missed the line 259.

Reply:
done